### ISSUE TITLE
Support running queries on a JPA EntityManager

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#6543]
For a tighter jOOQ / JPA integration, we should support running queries on a JPA EntityManager. There are different potential modes:

- Fetching projections
  - This could be done by extracting / unwrapping a JDBC Connection from the EntityManager and running the query as always. EclipseLink allows for unwrapping the connection through `em.unwrap()`. Hibernate requires proprietary API for this.
  - Alternatively, we can run a native query on the EntityManager and map the resulting `Object[]` to type safe jOOQ records
- Fetching entities
  - This would be done by running native queries on the EntityManager by passing the entity class
  - Alternatively, the obscure `SqlResultSetMapping` could be used (although, it's probably not widely adopted)
  - Perhaps, we could also resort to using vendor-specific mapping capabilities, such as Hibernate's [`ResultTransformer`](https://docs.jboss.org/hibernate/orm/current/javadocs/org/hibernate/transform/ResultTransformer.html) or [`NativeQueryImplementor.addEntity()`](https://docs.jboss.org/hibernate/orm/current/javadocs/org/hibernate/query/spi/NativeQueryImplementor.html#addEntity-java.lang.Class-) etc.

There are a few open questions, including

- What happens to the `ExecuteListener` SPI semantics (which depends on the JDBC Connection)
- What happens to the `Converter` and `Binding` semantics - we should probably emulate them. This would probably be done via #4886
- ~~What happens to `RecordListener` semantics. This will be implemented through #6544~~ (in fact, this functionality will be implemented through a new `fetchEntities()` API, so `RecordListener` semantics is not affected)

**Comment by lukaseder** [comment]
This needs much more thorough thinking and design - we don't want to postpone jOOQ 3.10 any further. Let's make this a high priority in 3.11, though.

**Comment by lukaseder** [comment]
The idea is interesting, academically. But practically, there aren't really many benefits to doing this. Most jOOQ users want to run queries with jOOQ directly, benefitting from all of jOOQ's features, including e.g. `MULTISET`. These wouldn't be available, if the query is run on a JPA `EntityManager`, just like `Binding` and `Converter` might have problems. We made them work to some extent on R2DBC, but if anything at all, the R2DBC integration has shown the complexity of supporting a second runtime in addition to JDBC. But unlike R2DBC support, which was implemented due to popular demand, this here has been asked for by no one.
